### PR TITLE
Improve pulse waveform validation error messagesFixed pulse validation edge case

### PR DESCRIPTION
**Pull request by Janvee31**
## Summary

Improved the validation error messages in `pulse.py` to make them clearer and more user-friendly.

## Testing

- Ran targeted tests:
  - `pytest tests/test_pulse.py`
- Ran full test suite:
  - `pytest -q`

All tests passed successfully.

### CODE REVIEW

**Comment by a-corni** [comment]
Hello @Janvee31 ! Thanks for your contribution !

**Comment by a-corni** [comment]
Hello @Janvee31,
We have more CI tests than pytest. We notably check the "style" of the package, and would like you to run "black ." on your code to fix the "style" check. Please, have a look at [our contributing section](https://github.com/pasqal-io/Pulser/blob/develop/CONTRIBUTING.md#continuous-integration-requirements)

File: pulser-core/pulser/pulse.py [hunk 1]
```diff
@@ -106,9 +106,9 @@ def __init__(
             )
         if np.any(amplitude.samples.as_array(detach=True) < 0):
             raise ValueError(
-                "All samples of an amplitude waveform must be "
-                "greater than or equal to zero."
-            )
+                f"All samples of an amplitude waveform must be >= 0; "
```
Comment (a-corni) [inline pulser-core/pulser/pulse.py:109]: flake8 is complaining because this f-string doesn't have any value to fill in. Delete the f to pass the tests.

File: pulser-core/pulser/pulse.py [hunk 2]
```diff
@@ -106,9 +106,9 @@ def __init__(
...
+                f"found invalid values in amplitude waveform: {amplitude}."
```
Comment (a-corni) [inline pulser-core/pulser/pulse.py:110]: I find the suggestion of @HGSilveri better:
```suggestion
                f"found negative values in amplitude waveform: {amplitude}."
```

**Comment by a-corni** [comment]
This tackles partially #1057 (only for pulse waveform).

**Comment by HGSilveri** [comment]
Hey @Janvee31, it's been a while since this PR has been idle. Are you planning on picking it back up soon?

**Comment by a-corni** [comment]
Closing this PR as completed by #1093
